File: test_data/code/java/1009-complement-of-base-10-integer.java
class Solution {
    public int bitwiseComplement(int n) {
        int temp = n, sum = 0, keep = 1;
        while (temp > 0) {
            temp = temp >> 1;
            sum = sum + keep * (1 & (n ^ temp));
            keep *= 2;
        }
        return sum;
    }
}

File: test_data/code/python/1009-complement-of-base-10-integer.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1
        binary = bin(n)[2:]  # Remove the '0b' prefix
        complement = ''
        for bit in binary:
            if bit == '0':
                complement += '1'
            else:
                complement += '0'
        return int(complement, 2)